transcom/mymove
# MB-13754: Working client logging

## [Jira ticket](https://dp3.atlassian.net/browse/MB-13754)

## Summary

* Rework milmoveLogger to be compliant with @opentelemetry/api
* add componentDidCatch to SystemAdmin page to log errors

## How to test locally

### Testing with console logging

1. Start the client: `make client_run`. Note that the `.envrc` changes ensure the client logs to the console
2. Start the server: `make server_run`.
3. Go to `http://milmovelocal:3000`
4. Open the dev tools and look at the console. You should see something like:
```
Operation users.showLoggedInUser failed: Error: Unauthorized (401)
(anonymous) @ consoleLogger.ts:49
```

### Testing sending logs to the server (like we will do in deployed environments)

1. Start the client: `REACT_APP_ERROR_LOGGING=otel make client_run`
2. Start the server: `SERVE_CLIENT_COLLECTOR=true make server_run`
3. Go to `http://milmovelocal:3000`
4. Look at the server logs. You should see something like:
```
2023-08-08T21:29:34.857Z	INFO	handlers/client_log_collector.go:48	client log upload stats	
{"git_branch": "adh-MB-13754-browser-logging", "git_commit": "9c073c46b1cf06a12aedc0ec08f325c7e5dac086", 
"source": "client_stats", "app": "my", "logEntryCount": 1, "droppedLogCount": 0, "failedSendCount": 0, 
"failedTimerCount": 0}
2023-08-08T21:29:34.857Z	INFO	handlers/client_log_collector.go:59	client log entry	
{"git_branch": "adh-MB-13754-browser-logging", "git_commit": "9c073c46b1cf06a12aedc0ec08f325c7e5dac086",
 "source": "client_log_entry", "app": "my", "logLevel": "error", "args": ["Operation users.showLoggedInUser failed: Error:
 Unauthorized (401)"]}
```

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.
